dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -240,18 +226,28 @@ DecodeResult NvJpeg2000DecoderInstance::DecodeImplTask(int thread_idx,
 
   CUDA_CALL(cudaEventSynchronize(ctx.decode_event));
 
-  // nvJPEG2000 decodes into planar layout
-  bool is_conversion_needed = ctx.shape[2] > 1 || ctx.bpp == 16;
+  DALIImageType format = ctx.shape[2] == 1 ? DALI_GRAY : DALI_RGB;
+  bool is_processing_needed =
+    ctx.shape[2] > 1 ||  // nvJPEG2000 decodes into planar layout
```

Comment:
Just a question. If we have planar format, what layout is the shape here? Is it still HWC? I would expect CHW and channels residing in [0].